### DISCUSSION TITLE
ZBBridge upgrade Zigbee devices

### DISCUSSION THREAD

**Discussion by ghwaldo** [arendst/Tasmota#21787] [Support request]
Hello,

i use some TRVZB devices, works fine so far.
But how can i upgrade firmware ? Anyone here doing such things ? 
Or do i have to use another zigbee bridge with stock firmware ?

**Comment by Jason2866** [comment]
Tasmota does not support upgrading zigbee firmwares

**Reply by sfromis** [reply]
I believe that you can do such device firmware upgrades with Zigbee2MQTT, if you temporarily pair with that instead.

**Reply by ghwaldo** [reply]
> Tasmota does not support upgrading zigbee firmwares

I know, but thought someone here could show me a way of doing this.



> I believe that you can do such device firmware upgrades with Zigbee2MQTT, if you temporarily pair with that instead.

But that would mean i have to setup another environment with zigbee2mqtt software / hardware !?

**Reply by sfromis** [reply]
The reason why I mentioned Zigbee2MQTT as an option is that is is feasible without more hardware. You'd need to install the project (obviously), like on an existing server, and temporarily have the bridge communicate with it instead of the Tasmota Zigbee2Tasmota protocol.

**Reply by ghwaldo** [reply]
I now have setup zigbee2mqtt on the pi and use tasmota on the zbbridge as a device in zigbee2mqt via tcp.//....

It works with updates.

Currently i'm not thinking about reverting back as i have to pair the devices again. It works, currently i`m not using any rules or something to build processes between devices.
